src/server/svr_func.c

```diff
@@ -7212,19 +7212,20 @@ void
 set_attr_svr(attribute *pattr, attribute_def *pdef, char *value)
 {
 	attribute tempat;
+	int rc;
 
 	if (pattr == NULL || pdef == NULL) {
-		(void)sprintf(log_buffer,  "Invalid pointer to attribute or its definition");
-		log_err(errno, __func__, log_buffer);
+		snprintf(log_buffer, sizeof(log_buffer), "Invalid pointer to attribute or its definition");
+		log_err(-1, __func__, log_buffer);
 		return;
 	}
 	clear_attr(&tempat, pdef);
-	if (pdef->at_decode(&tempat, pdef->at_name, NULL, value) != 0) {
-		(void)sprintf(log_buffer,  "decode of %s failed", pdef->at_name);
-		log_err(errno, __func__, log_buffer);
-	} else if (pdef->at_set(pattr, &tempat, SET) != 0) {
-		(void)sprintf(log_buffer, "set of %s failed", pdef->at_name);
-		log_err(errno, __func__, log_buffer);
+	if ((rc = pdef->at_decode(&tempat, pdef->at_name, NULL, value)) != 0) {
+		snprintf(log_buffer,  sizeof(log_buffer), "decode of %s failed", pdef->at_name);
+		log_err(rc, __func__, log_buffer);
+	} else if ((rc = pdef->at_set(pattr, &tempat, SET)) != 0) {
+		(void)snprintf(log_buffer, sizeof(log_buffer), "set of %s failed", pdef->at_name);
```

Comment:
Please remove the cast to void for this call as well.

Reply:
Done